include/deal.II/numerics/fe_field_function.templates.h

```diff
@@ -140,7 +141,14 @@ namespace Functions
     FEValues<dim> fe_v(mapping, cell->get_fe(), quad,
                        update_gradients);
     fe_v.reinit(cell);
-    fe_v.get_function_gradients(data_vector, gradients);
+
+    // Unfortunately we still need a temporary argument as we want to
+    // evaluate a gradient of a (generally) multicomponent function at
+    // a single quadrature point.
+    std::vector< std::vector<Tensor<1,dim,number> > > vgrads
+    (1,  std::vector<Tensor<1,dim,number> >(n_components) );
+    fe_v.get_function_gradients(data_vector, vgrads);
+    gradients = std::vector<Tensor<1,dim,number> >(vgrads[0].begin(), vgrads[0].end());
```

Comment:
we call proper function, it's all good. 

The point is, that the first `std::vector<>` has to do with the number of quadrature points, which is 1 always here. 
Whereas the second `std::vector<>` has to do with the number of components in the function. That's why you need to have that temporary vector, which I tried to explain in the comment. 

If you look in the original code https://github.com/dealii/dealii/pull/2219#discussion_r53622136 , 
the comment was actually wrong! We need that auxiliary variable irrespectively of data types.